kernel/src/engine/default/parquet.rs

```diff
@@ -27,6 +34,66 @@ pub struct DefaultParquetHandler<E: TaskExecutor> {
     readahead: usize,
 }
 
+/// Metadata of a parquet file, currently just includes the file metadata but will expand to
+/// include file statistics and other metadata in the future.
+#[derive(Debug)]
+pub struct ParquetWriteMetadata {
+    file_meta: FileMeta,
+}
+
+impl ParquetWriteMetadata {
+    pub fn new(file_meta: FileMeta) -> Self {
+        Self { file_meta }
+    }
+
+    // convert ParquetMetadata into a record batch which matches the 'write_metadata' schema
+    fn as_record_batch(
+        &self,
+        partition_values: HashMap<String, String>,
+        data_change: bool,
+    ) -> DeltaResult<Box<dyn EngineData>> {
+        let ParquetWriteMetadata { file_meta } = self;
+        let FileMeta {
+            location,
+            last_modified,
+            size,
+        } = file_meta;
+        let write_metadata_schema = crate::transaction::get_write_metadata_schema();
+
+        // create the record batch of the write metadata
+        let path = Arc::new(StringArray::from(vec![location.to_string()]));
+        let key_builder = StringBuilder::new();
+        let val_builder = StringBuilder::new();
+        let names = MapFieldNames {
+            entry: "key_value".to_string(),
+            key: "key".to_string(),
+            value: "value".to_string(),
```

Comment:
interesting - I'm not really sure what's 'better form' - I usually err on the side of `to_string()` since that seems to be more explicit about an allocation being performed?